test/integration/test-amp-analytics.js

```diff
@@ -66,7 +66,216 @@ describe.configure().skipIfPropertiesObfuscated().run('amp' +
+      `,
+    extensions: ['amp-analytics'],
+  }, env => {
+    let browser;
+
+    beforeEach(() => {
+      browser = new BrowserController(env.win);
+    });
+
+    it('should send request', () => {
+      const reqPromise = RequestBank.withdraw().then(req => {
+        expect(req.url).to.equal('/?f=hello%20world&b=2');
+      });
+      browser.click('a');
+      return reqPromise;
+    });
+  });
+
+  describes.integration('scroll trigger', {
+    body: `
+      <amp-analytics>
+        <script type="application/json">
+        {
+          "requests": {
+            "endpoint": "${RequestBank.getUrl()}"
+          },
+          "triggers": {
+            "scroll": {
+              "on": "scroll",
+              "request": "endpoint",
+              "scrollSpec": {
+                "verticalBoundaries": [70]
+              },
+              "extraUrlParams": {
+                "scrollTop": "\${scrollTop}",
+                "scrollHeight": "\${scrollHeight}"
+              }
+            }
+          }
+        }
+        </script>
+      </amp-analytics>
+      <div class="block" style="height: 100vh; background: red">
+        1st viewport
+      </div>
+      <div class="block" style="height: 100vh; background: blue">
+        2nd viewport
+      </div>
+      `,
+    extensions: ['amp-analytics'],
+  }, env => {
+    let browser;
+
+    beforeEach(() => {
+      browser = new BrowserController(env.win);
+    });
+
+    it('should trigger on scroll', () => {
+      const reqPromise = RequestBank.withdraw().then(req => {
+        expect(req.url).to.equal('/?scrollTop=75&scrollHeight=300');
+      });
+      // verticalBoundaries is set to 70%
+      // (windowHeight + scrollTop) / scrollHeight = (150 + 75) / 300 = 75%
+      // so scrolling 75px guarantees a triggering
+      browser.scrollTo(75);
+      return reqPromise;
+    });
+  });
+
+  describes.integration('element visible trigger', {
+    body: `
+      <amp-analytics>
+        <script type="application/json">
+        {
+          "requests": {
+            "endpoint": "${RequestBank.getUrl()}"
+          },
+          "triggers": {
+            "visible": {
+              "on": "visible",
+              "request": "endpoint",
+              "visibilitySpec": {
+                "selector": "amp-img",
+                "selectionMethod": "scope",
+                "visiblePercentageMin": 50,
+                "totalTimeMin": 500
+              },
+              "extraUrlParams": {
+                "timestamp": "\${timestamp}",
+                "loadTimeVisibility": "\${loadTimeVisibility}",
+                "maxVisiblePercentage": "\${maxVisiblePercentage}",
+                "totalVisibleTime": "\${totalVisibleTime}"
+              }
+            }
+          }
+        }
+        </script>
+      </amp-analytics>
+      <div class="block" style="height: 100vh; background: red">
+        1st viewport
+      </div>
+      <amp-img layout="fixed" width="300" height="150"
```

Comment:
same question, can we always be sure that the image can be 50% visible, that the window size is twice more than image's size.